app/src/main/java/com/example/intandem/fragments/PostsFragment.java

```diff
@@ -207,18 +354,82 @@ public void done(List<Post> posts, ParseException e) {
                         allPosts.addAll(posts);
-                        adapter.notifyDataSetChanged();
+                        if (maxDistance == -1) {
+                            adapter.notifyDataSetChanged();
+                            return;
+                        }
+                        // FILTERING BY DISTANCE
+                        filteredDistancePosts.clear();
+
+                        Retrofit retrofit = new Retrofit.Builder()
+                                .baseUrl(BASE_URL)
+                                .addConverterFactory(GsonConverterFactory.create())
+                                .build();
+                        StringBuilder destinations = new StringBuilder();
+                        for (int i = 0; i < allPosts.size() - 1; i++) {
+                            CustomPlace customPlace = allPosts.get(i).getCustomPlace();
+                            try {
+                                customPlace.fetchIfNeeded();
+                                destinations.append("place_id:").append(customPlace.getGPlaceId()).append("|");
+                            } catch (ParseException ex) {
+                                Log.e(TAG, ex.toString());
+                                ex.printStackTrace();
+                            }
+                        }
+                        CustomPlace customPlace = allPosts.get(allPosts.size() - 1).getCustomPlace();
+                        try {
+                            customPlace.fetchIfNeeded();
+                            destinations.append("place_id:").append(customPlace.getGPlaceId());
+                        } catch (ParseException ex) {
+                            Log.e(TAG, ex.toString());
+                            ex.printStackTrace();
+                        }
+
+                        GoogleMapsService googleMapsService = retrofit.create(GoogleMapsService.class);
+                        googleMapsService.getDistanceSearchResult(latitude + "," + longitude,
+                                destinations.toString(),
+                                "driving",
+                                "en",
+                                BuildConfig.MAPS_API_KEY).enqueue(new Callback<DistanceSearchResult>() {
+                            @Override
+                            public void onResponse(Call<DistanceSearchResult> call, Response<DistanceSearchResult> response) {
+                                DistanceSearchResult distanceSearchResult = response.body();
+                                Log.i(TAG, "success getting all the distances");
+                                List<DistanceInfo> elements = distanceSearchResult.getRows().get(0).getElements();
+                                for (int i = 0; i < elements.size(); i++) {
+                                    Log.d(TAG, "" + elements.get(i).getDistance().getValue() / 1000.0);
+                                    if ((elements.get(i).getDistance().getValue() / 1000.0) <= maxDistance) {
```

Comment:
Since we need `elements.get(i).getDistance().getValue() / 1000.0` twice, (or more i think i see more uses below) its best to put that in a variable and use that instead.